aten/src/ATen/native/cuda/ForeachBinaryOpScalar.cu

```diff
@@ -105,7 +133,23 @@ std::vector<Tensor> foreach_tensor_sub_scalar_kernel_cuda(TensorList tensors, co
         return at::native::foreach_tensor_sub_scalar_kernel_slow(tensors, scalar);
     }
 
-    return foreach_binary_op<std::minus>(tensors, scalar);
+    return AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(kBool, kHalf, kBFloat16, tensors[0].scalar_type(), "foreach_binary_op_scalar_cuda", [&]() {
+        return foreach_binary_op<scalar_t, std::minus>(tensors, scalar);
+    });
 }
 
+// std:: does not have clamp functors
+template <typename T>
+struct clamp_min {
+    __device__ T operator()(const T& a, const T& b) const { return a < b ? b: a; }
+};
+
+template <typename T>
+struct clamp_max {
+    __device__ T operator()(const T& a, const T& b) const { return a > b ? b: a; }
+};
+
+FOREACH_BINARY_OP_SCALAR(all_types_bool_half_bfloat16, clamp_min, clamp_min, false);
```

Comment:
We already have foreach_minimum and foreach_maximum ops that do the same thins as tensor clamp_min/max overloads, now we are compiling these kernels twice. Please think about how you can avoid duplication - dispatch minimum/maximum kernels to clamp? The other way around?

Reply:
I think using this `clamp_max/min` implementation for `foreach_minimum/maximum` works. This would remove the custom code in `ForeachPointwiseOp.cu`.

The `foreach_maximum/minimum` can then also be dispatched to clamp for the scalar and scalarlist cases, which don't exist for the current `foreach_minimum/maximum`.

So `foreach_minimum` would just be an alias of `foreach_clamp_max`. We could also create a `foreach_clip_max` alias since both `torch.clamp` and `torch.clip` exist.

Does that make sense?

Reply:
Yeah that makes sense. Note that currently regular `minimum` and `maximum` don't support scalar overloads, but given that all kernels are compiled for `foreach_clamp*` anyway, it doesn't make sense to artificially restrict `foreach_minimum/maximum`

Reply:
Done